src/libraries/System.Private.CoreLib/src/System/IO/Path.Windows.cs

```diff
@@ -208,25 +208,56 @@ static uint GetTempPathW(int bufferLen, ref char buffer)
         // name on disk.
         public static string GetTempFileName()
         {
-            var tempPathBuilder = new ValueStringBuilder(stackalloc char[PathInternal.MaxShortPath]);
-
-            GetTempPath(ref tempPathBuilder);
-
-            var builder = new ValueStringBuilder(stackalloc char[PathInternal.MaxShortPath]);
-
-            uint result = Interop.Kernel32.GetTempFileNameW(
-                ref tempPathBuilder.GetPinnableReference(), "tmp", 0, ref builder.GetPinnableReference());
-
-            tempPathBuilder.Dispose();
-
-            if (result == 0)
-                throw Win32Marshal.GetExceptionForLastWin32Error();
-
-            builder.Length = builder.RawChars.IndexOf('\0');
+            // Avoid GetTempFileNameW because it is limited to 0xFFFF possibilities, which both
+            // means that it may have to make many attempts to create the file before
+            // finding an unused name, and also that if an app "leaks" such temp files,
+            // it can prevent GetTempFileNameW succeeding at all.
+            //
+            // To make this a little more robust, generate our own name with more
+            // entropy. We could use GetRandomFileName() here, but for consistency
+            // with Unix and to retain the ".tmp" extension we will use the "tmpXXXXXX.tmp" pattern.
+            // Using 32 characters for convenience, that gives us 32^^6 ~= 10^^9 possibilities,
+            // but we'll still loop to handle the unlikely case the file already exists.
+
+            byte* bytes = stackalloc byte[KeyLength];
```

Comment:
Does `KeyLength` here bind to the const on line 228? Or the const here:

https://github.com/dotnet/runtime/blob/62948580b4d3547473cb567bff6cc61308136d81/src/libraries/System.Private.CoreLib/src/System/IO/Path.cs#L25

Reply:
the nearby one (according to VS)

Reply:
That actually blows my mind, since it is:

1. Declared after this line
2. Inside another scope

I wonder if that is just a thing for `const`s.

Reply:
sharplab.io disagrees.

https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQMwAJboMLoN7LpHoAOATgJYBuAhgC4Cm6AxgPYB2AznehezwGkGATwAyDdgHM6AC3QBedAA4A3IWLqiadAFcuNAGZMoAFnQBZABQBKfJuJEARsMYAqdM8acF6bjWYA1jQANsGszB4uDADaQmIS0jIAumpIDg4EaenZbFw8fIIi4lKyPiap2dlQcACclnHFidYVlQC+9q0aSK1AA===

This is unsafe code. If you get it wrong, we may not know until it is too late. This needs to be written in more obvious way - const declared before first use, and use non-colliding name.

Reply:
oh - I see your point. will fix.

Reply:
> That actually blows my mind

FYI it was my bad looking at the wrong line, not some strange scoping thing I thought wrongly about 😄